collect/collect.go

```diff
@@ -346,7 +333,7 @@ func isRootSpan(sp *types.Span) bool {
 
 // sendASAP sends a trace as soon as possible, aka no delay.
 func (i *InMemCollector) sendASAP(trace *types.Trace) {
-	go i.pauseAndSend(0, trace)
+	trace.SetDeadline(0)
```

Comment:
I think this will have to change to actually trigger sending the trace. `sendASAP` is invoked when a new trace is added to the buffer of pending traces but the buffer is close enough to full that an unsent trace is ejected from the ring. Setting the deadline on the trace won't trigger it getting sent since it's already been ejected from the set of traces that will be checked by the ticker.

Open question whether this sending should be in a goroutine or synchronous. Since it's in the path of `processSpan` (aka incoming traffic) I think nonblocking is better? Open to suggestions.

```suggestion
	go func(){
		// TODO fix FinishTime to be the time of the last span + its duration
		trace.FinishTime = time.Now()
		i.send(trace)
	){}
```

Reply:
A lot of the code in `send` modifies traces which may cause some issue. I think this is a good idea to do it async but perhaps if we want to go down this route then having a separate go routine to send traces is a better idea than creating arbitrary go routines like this?